core/modules/file/views/views_handler_field_file_icon.inc

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * @file
+ * Definition of views_handler_field_file_icon.
+ */
+
+/**
+ * @ingroup views_field_handlers
+ */
+class views_handler_field_file_icon extends views_handler_field {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function option_definition() {
+    $options = parent::option_definition();
+    $options['icon_size'] = array('default' => 32);
+    $options['display_type'] = array('default' => 'svg');
+    $options['add_alt_text'] = array('default' => TRUE);
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function options_form(&$form, &$form_state) {
+    parent::options_form($form, $form_state);
+    $form['icon_size'] = array(
+      '#type' => 'number',
+      '#title' => t('Icon size'),
+      '#min' => 8,
+      '#max' => 300,
+      '#default_value' => $this->options['icon_size'],
+      '#field_suffix' => 'px',
+    );
+    $form['display_type'] = array(
+      '#type' => 'radios',
+      '#title' => t('Display type'),
+      '#options' => array(
+        'svg' => t('Inline <code>&lt;svg&gt;</code> tag'),
+        'img' => t('Regular <code>&lt;img&gt;</code> tag'),
+      ),
+      '#default_value' => $this->options['display_type'],
+    );
+    $form['display_type']['svg']['#description'] = t('Inline SVG images are usually more efficient and easier to style.');
+    $form['display_type']['img']['#description'] = t('Regular image tags can be more efficient when showing the same icon many times.');
+    $form['add_alt_text'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Add alternative text'),
+      '#default_value' => $this->options['add_alt_text'],
+      '#description' => t('If alternative text is omitted, this icon will be marked as decorative only.'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function render($values) {
+    $uri = $this->get_value($values);
+
+    // If there is an extension.
+    if ($extension = pathinfo($uri, PATHINFO_EXTENSION)) {
+      $icon_name = $this->getIconName($uri);
+      if ($this->options['display_type'] == 'img') {
+        $icon = $this->renderImg($icon_name, $extension);
+      }
+      else {
+        $icon = $this->renderSvg($icon_name, $extension);
+      }
```

Comment:
```php
if (!$this->options['add_alt_text']) {
  $attributes['aria-hidden'] = "true");
}
```
Since it is the same for both, we could add it here?